src/App.tsx

```diff
@@ -1,10 +1,59 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/Filter';
 
-const USER_ID = 0;
+const USER_ID = 6701;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [statusFilter, setStatusFilter] = useState<FilterType>(FilterType.All);
+  const [loadingError, setLoadingError] = useState(false);
+
+  getTodos(USER_ID)
+    .then(res => {
+      setTodos(res);
+    })
+    .catch(() => {
+      setLoadingError(true);
+      setTodos([]);
+    });
+
+  const filterTodos = (filter:FilterType) => {
+    switch (filter) {
+      case FilterType.Active:
+        return (todos.filter(todo => !todo.completed));
+
+      case FilterType.Completed:
+        return (todos.filter(todo => todo.completed));
+
+      default:
+        return [...todos];
+    }
+  };
+
+  const visibleTodos:Todo[] = useMemo(
+    () => filterTodos(statusFilter), [statusFilter, todos],
+  );
+
+  const clickAllHandler = () => {
+    setStatusFilter(FilterType.All);
+  };
+
+  const clickActiveHandler = () => {
+    setStatusFilter(FilterType.Active);
+  };
+
+  const clickCompletedHandler = () => {
+    setStatusFilter(FilterType.Completed);
+  };
```

Comment:
Why do you need separate handlers for it? You can create one function and pass the filter type as an argument